src/backend/dx12/src/device.rs

```diff
@@ -1877,10 +1938,34 @@ impl d::Device<B> for Device {
             }
         }
 
+        // See [`introspect_spirv_vertex_semantic_remapping`] for details of why this is needed.
+        let semantics: Vec<_> = attributes.iter().map(|attrib| {
```

Comment:
similarly here, why don't we compute this from within the other loop, instead of collecting into `semantics` vector?